### DISCUSSION TITLE
Do you have any plan to expand the scope to CPU?

### DISCUSSION THREAD

**Discussion by mengniwang95** [huggingface/text-generation-inference#1118] [General]
Hi, I find this repo mainly focus on LLM inference on GPUs currently.  Do you have any plan to expand the scope to CPU?

Our team develop the [Intel® Extension for Transformers](https://github.com/intel/intel-extension-for-transformers), which is an innovative toolkit to accelerate Transformer-based models on Intel platforms, in particular effective on 4th Intel Xeon Scalable processor [Sapphire Rapids](https://www.intel.com/content/www/us/en/products/docs/processors/xeon-accelerated/4th-gen-xeon-scalable-processors.html) (codenamed Sapphire Rapids). The toolkit provides the below key features:

- Seamless user experience of model compressions (include RTN, AWQ, GPTQ, bitsandbytes and other our own algorithms in the future for weight-only quantization) on Transformer-based models by extending [Hugging Face transformers](https://github.com/huggingface/transformers) APIs and leveraging [Intel® Neural Compressor](https://github.com/intel/neural-compressor)

- Advanced software optimizations and unique compression-aware runtime.

- Optimized Transformer-based model packages.

- [NeuralChat](https://github.com/intel/intel-extension-for-transformers/blob/main/intel_extension_for_transformers/neural_chat), a customizable chatbot framework to create your own chatbot within minutes by leveraging a rich set of plugins and SOTA optimizations.

- [Inference](https://github.com/intel/intel-extension-for-transformers/blob/main/intel_extension_for_transformers/llm/runtime/graph) of Large Language Model (LLM) in pure C/C++ with weight-only quantization kernels.

We want to make some contributions to LLM ecosystem sincerely and TGI is a really popular project. So, is there any chance to integrate part of our work into TGI?

Thanks

**Comment by HydeAndGeek** [comment]
Any updates on this? Looking for one click installers to connect multiple llm models for testing on Intel iris xe gpu on a Lenovo i7 seems impossible to get anything running even 50% to what the other nvda are capable of running. I just got this thing  as a gift otherwise would replace it. Thanks in advance
